src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx

```diff
@@ -873,7 +876,11 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
     runeTxFeeCryptoBaseUnit,
   ])
 
-  const handleApprove = useCallback(() => mutate(undefined), [mutate])
+  const handleApprove = useCallback(async () => {
+    if (!assetId) return
+    await checkLedgerAppOpenIfLedgerConnected(fromAssetId(assetId).chainId)
```

Comment:
option to leverage `onMutate` to tie this logic directly to the mutation

Reply:
Prefer to keep it that way as that would make us destructure the `mutationFn` from `reactQueries.mutations.approve`  which looks a bit flaky IMO i.e:

```diff
diff --git a/src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx b/src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx
index 9ceb939395..bcbe779ced 100644
--- a/src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx
+++ b/src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx
@@ -591,23 +591,30 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
     return serializeTxIndex(poolAssetAccountId, approvalTxId, poolAssetAccountAddress)
   }, [approvalTxId, poolAssetAccountAddress, poolAssetAccountId])
 
+  const { mutationKey, mutationFn } = reactQueries.mutations.approve({
+    assetId: poolAsset?.assetId,
+    spender: poolAssetInboundAddress,
+    amountCryptoBaseUnit: toBaseUnit(
+      actualAssetDepositAmountCryptoPrecision,
+      poolAsset?.precision ?? 0,
+      BigNumber.ROUND_UP,
+    ),
+    wallet: wallet ?? undefined,
+    from: poolAssetAccountId ? fromAccountId(poolAssetAccountId).account : undefined,
+    accountNumber: poolAssetAccountNumber,
+  })
+
   const {
     mutate,
     isPending: isApprovalMutationPending,
     isSuccess: isApprovalMutationSuccess,
   } = useMutation({
-    ...reactQueries.mutations.approve({
-      assetId: poolAsset?.assetId,
-      spender: poolAssetInboundAddress,
-      amountCryptoBaseUnit: toBaseUnit(
-        actualAssetDepositAmountCryptoPrecision,
-        poolAsset?.precision ?? 0,
-        BigNumber.ROUND_UP,
-      ),
-      wallet: wallet ?? undefined,
-      from: poolAssetAccountId ? fromAccountId(poolAssetAccountId).account : undefined,
-      accountNumber: poolAssetAccountNumber,
-    }),
+    mutationKey,
+    mutationFn: async (_: void) => {
+      if (!assetId) throw new Error('assetId is required')
+      await checkLedgerAppOpenIfLedgerConnected(fromAssetId(assetId).chainId)
+      return mutationFn(undefined)
+    },
     onSuccess: (txId: string) => {
       setApprovalTxId(txId)
     },
@@ -876,11 +883,7 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
     runeTxFeeCryptoBaseUnit,
   ])
 
-  const handleApprove = useCallback(async () => {
-    if (!assetId) return
-    await checkLedgerAppOpenIfLedgerConnected(fromAssetId(assetId).chainId)
-    mutate(undefined)
-  }, [assetId, checkLedgerAppOpenIfLedgerConnected, mutate])
+  const handleApprove = useCallback(() => mutate(undefined), [mutate])
 
   const handleSubmit = useCallback(() => {
     if (isApprovalRequired) return handleApprove()
```

Reply:
Was thinking more along the lines of:

```
diff --git a/src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx b/src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx
index 9ceb939395..81e9916893 100644
--- a/src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx
+++ b/src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx
@@ -608,6 +608,10 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
       from: poolAssetAccountId ? fromAccountId(poolAssetAccountId).account : undefined,
       accountNumber: poolAssetAccountNumber,
     }),
+    onMutate: async () => {
+      if (!poolAsset?.chainId) return
+      await checkLedgerAppOpenIfLedgerConnected(poolAsset.chainId)
+    },
     onSuccess: (txId: string) => {
       setApprovalTxId(txId)
     },
@@ -876,12 +880,6 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
     runeTxFeeCryptoBaseUnit,
   ])
 
-  const handleApprove = useCallback(async () => {
-    if (!assetId) return
-    await checkLedgerAppOpenIfLedgerConnected(fromAssetId(assetId).chainId)
-    mutate(undefined)
-  }, [assetId, checkLedgerAppOpenIfLedgerConnected, mutate])
-
   const handleSubmit = useCallback(() => {
     if (isApprovalRequired) return handleApprove()
     if (isSweepNeeded) return history.push(AddLiquidityRoutePaths.Sweep)
```